definitions/apply-mapping/apply-mapping.yaml

```diff
@@ -0,0 +1,59 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: apply-mapping
+spec:
+  params:
+    - name: applicationSnapshot
+      type: string
+      description: The ApplicationSnapshot in JSON format to apply the mapping to
+    - name: extraConfigPath
+      type: string
+      description: The path to the config file containing the mapping
+      default: ""
+  workspaces:
+    - name: config
+      description: The workspace where the extra config file containing the mapping resides
+  results:
+    - name: applicationSnapshot
+      description: |
+        The AppicationSnapshot in JSON format with the mapping applied. If there is no mapping key in the
+        extraConfigPath, this result will be the same as the original applicationSnapshot parameter.
+  steps:
+    - name: apply-mapping
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        CONFIG_FILE="$(workspaces.config.path)/$(params.extraConfigPath)"
+        if [ ! -f "${CONFIG_FILE}" ] ; then
+            echo "No valid config file was provided."
+            echo '$(params.applicationSnapshot)' | tee $(results.applicationSnapshot.path)
+            exit 0
+        fi
+        MAPPING_EXISTS=$(yq 'has("mapping")' "${CONFIG_FILE}")
+        if [ "${MAPPING_EXISTS}" != "true" ] ; then
+            echo "Config file contains no mapping key."
+            echo '$(params.applicationSnapshot)' | tee $(results.applicationSnapshot.path)
+            exit 0
+        fi
+
+        # Convert applicationSnapshot to YAML so it matches the config file
+        echo '$(params.applicationSnapshot)' | yq -P '.' > snapshot.yaml
+
+        # Merge the mapping key in the config file with the images key in the snapshot based on component name
+        selector=".component" configPath=".mapping" snapshotPath=".images" yq eval-all '
+        (
+          (( (eval(strenv(configPath)) + eval(strenv(snapshotPath)))  | .[]
+          | {(eval(strenv(selector))):  .}) as $item ireduce ({}; . * $item )) as $uniqueMap
+          | ( $uniqueMap  | to_entries | .[]) as $item ireduce([]; . + $item.value)
+        ) as $mergedArray
+        | select(fi == 0) | (eval(strenv(configPath))) = $mergedArray
+        ' ${CONFIG_FILE} snapshot.yaml > filtered_snapshot.yaml
+
+        # Convert yaml file back to JSON to save as a task result
+        yq -o=json -I=0 '.' filtered_snapshot.yaml | tee $(results.applicationSnapshot.path)
+
+        rm -f snapshot.yaml
```

Comment:
We could probably use a trap to delete the file even if something in the script fails. I think it would be something like:
```
trap 'rm -f snapshot.yaml` EXIT
```
at the beginning of the file.